docs: add message about search in dev mode (#2239)
vuejs/vitepress: docs/reference/default-theme-search.md

## Local Search

::: info

> You need to build and preview/deploy your site for search to work. It currently does not work in dev.

Author: ```suggestion
You need to build and preview/deploy your site for local search to work.
```
